mlflow/server/js/src/experiment-tracking/components/CompareRunView.js

```diff
@@ -230,6 +336,13 @@ export class CompareRunView extends Component {
     );
   }
 
+  switchNonDiffRowsDisplay(blockClass, showDiffOnly) {
+    const nonDiffRows = document.querySelectorAll(`.compare-table .${blockClass} .non-diff-row`);
+    for (let index = 0; index < nonDiffRows.length; ++index) {
+      nonDiffRows[index].style.display = showDiffOnly ? 'none' : '';
+    }
+  }
```

Comment:
Instead of mutating the style, can we define a state and hide non-diff rows if it's true like:

```js
const rowsToRender = hideNonDiffRows ? rows.filter(...) : rows;
return render(rowsToRender)
```